source/common/common/mem_block_builder.h

```diff
@@ -102,6 +102,16 @@ template <class T> class MemBlockBuilder {
     return std::move(data_);
   }
 
+  /**
+   * Returns the underlying storage as a raw pointer, clearing this.
+   *
+   * @return the transferred storage.
+   */
+  T* releasePointer() {
+    write_span_ = absl::MakeSpan(static_cast<T*>(nullptr), 0);
```

Comment:
nit: suggest writing this is release().release(), which avoids duplicating the write_span_ erasure logic. It will get inlined anyway in optimized builds, so there's no perf loss.